method.h

```diff
@@ -73,7 +73,7 @@ typedef struct rb_callable_method_entry_struct { /* same fields with rb_method_e
 #define METHOD_ENTRY_COMPLEMENTED_SET(me)    ((me)->flags |= IMEMO_FL_USER3)
 #define METHOD_ENTRY_CACHED(me)              ((me)->flags & IMEMO_FL_USER4)
 #define METHOD_ENTRY_CACHED_SET(me)          ((me)->flags |= IMEMO_FL_USER4)
-#define METHOD_ENTRY_INVALIDATED(me)         ((me)->flags & IMEMO_FL_USER5)
+#define METHOD_ENTRY_INVALIDATED(me)         (UNDEF_P((VALUE)(me)) || (me)->flags & IMEMO_FL_USER5)
```

Comment:
cc @tenderlove since you were worried this may add overhead to checking entries, apparently benchmarks indicate they don't?

Another possibility we mentioned with @peterzhu2118 was to have two `rb_gc_mark_weak`, one that would set to `Qundef` on sweep, that the other that would set to `0`.

That would however complexify the GC.

Reply:
Hum, I was assuming whatever the default value is is already checked elsewhere. No?

Reply:
Ah I see. Yeah I suppose `!= Qundef` and `!= 0` should have similarish perf.